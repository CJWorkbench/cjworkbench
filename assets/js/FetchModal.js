// Button that pops up a modal with fetch parameters

import React from 'react'
import { Button, Modal, ModalHeader, ModalBody, ModalFooter } from 'reactstrap'
import { Form, FormGroup, Label, Input } from 'reactstrap'


export default class FetchModal extends React.Component {
  constructor(props) {
    super(props);
    this.state = { modal: false };
    this.toggleModal = this.toggleModal.bind(this);
  }

  toggleModal() {
    this.setState({ modal: !this.state.modal });
  }

  render() {
    return (
      <span className="ml-2">
        <Button color='secondary' onClick={this.toggleModal}> {'\u23F1'} </Button>
        <Modal isOpen={this.state.modal} toggle={this.toggleModal} className={this.props.className}>
          <ModalHeader toggle={this.toggleModal}>External Data Settings</ModalHeader>
          <ModalBody>
            <FormGroup>
              <Label for="exampleSelect">Which version?</Label>
              <Input type="select" name="select" id="exampleSelect" className="mb-3">
                <option>Most Recent</option>
                <option value="" disabled="disabled">──────────</option>
                <option>April 28 at 4:40PM </option>
                <option>April 28 at 10:32AM </option>
                <option>April 26 at 6:06PM</option>
                <option>April 12 at 2:51PM</option>
              </Input>
              <Label for="exampleSelect2">Update when?</Label>
              <Input type="select" name="select" id="exampleSelect2">
                <option>Manual only</option>
                <option value="" disabled="disabled">──────────</option>
                <option>Every minute</option>
                <option>Every five minutes</option>
                <option>Hourly</option>
                <option>Daily</option>
                <option>Weekly</option>
              </Input>
            </FormGroup>
          </ModalBody>
          <ModalFooter>
            <Button color='primary' onClick={this.toggleModal}>Ok</Button>{' '}
            <Button color='secondary' onClick={this.toggleModal}>Cancel</Button>
          </ModalFooter>
        </Modal>
      </span>
    );
  }
}

