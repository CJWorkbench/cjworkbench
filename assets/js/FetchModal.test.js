import React from 'react';
import { shallow, render, mount } from 'enzyme';
import FetchModal from './FetchModal';

/*
it('renders correctly', () => {
  const tree = renderer.create(
    <FetchModal/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

*/


it('renders correctly', () => {
  const wrapper = shallow( <FetchModal /> );
  expect(wrapper).toMatchSnapshot();
});

it('opens when button pressed', () => {
  const wrapper = mount( <FetchModal /> );
  expect(wrapper.state().modal).toBe(false);
  wrapper.find('button').simulate('click');
  expect(wrapper.state().modal).toBe(true);
});

