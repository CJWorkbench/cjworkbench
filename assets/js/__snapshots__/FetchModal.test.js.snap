// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<span
  className="ml-2"
>
  <Button
    className="icon-Calendar"
    color="secondary"
    onClick={[Function]}
    tag="button"
  />
  <Modal
    autoFocus={true}
    backdrop={true}
    backdropTransitionTimeout={150}
    fade={true}
    isOpen={false}
    keyboard={true}
    modalTransitionTimeout={300}
    toggle={[Function]}
    zIndex={1050}
  >
    <ModalHeader
      tag="h4"
      toggle={[Function]}
      wrapTag="div"
    >
      External Data Settings
    </ModalHeader>
    <ModalBody
      tag="div"
    >
      <FormGroup
        tag="div"
      >
        <Label
          for="exampleSelect"
          tag="label"
        >
          Which version?
        </Label>
        <Input
          className="mb-3"
          id="exampleSelect"
          name="select"
          tag="p"
          type="select"
        >
          <option>
            Most Recent
          </option>
          <option
            disabled="disabled"
            value=""
          >
            ──────────
          </option>
          <option>
            April 28 at 4:40PM 
          </option>
          <option>
            April 28 at 10:32AM 
          </option>
          <option>
            April 26 at 6:06PM
          </option>
          <option>
            April 12 at 2:51PM
          </option>
        </Input>
        <Label
          for="exampleSelect2"
          tag="label"
        >
          Update when?
        </Label>
        <Input
          id="exampleSelect2"
          name="select"
          tag="p"
          type="select"
        >
          <option>
            Manual only
          </option>
          <option
            disabled="disabled"
            value=""
          >
            ──────────
          </option>
          <option>
            Every minute
          </option>
          <option>
            Every five minutes
          </option>
          <option>
            Hourly
          </option>
          <option>
            Daily
          </option>
          <option>
            Weekly
          </option>
        </Input>
      </FormGroup>
    </ModalBody>
    <ModalFooter
      tag="div"
    >
      <Button
        color="primary"
        onClick={[Function]}
        tag="button"
      >
        Ok
      </Button>
       
      <Button
        color="secondary"
        onClick={[Function]}
        tag="button"
      >
        Cancel
      </Button>
    </ModalFooter>
  </Modal>
</span>
`;
